### PR TITLE
SME and QE review requests for BXMSDOC-3281 Document support for multiple branches

### PR DESCRIPTION
**Pull request by michelehaglund**
Rendered output: http://file.rdu.redhat.com/~mhaglund/BXMSDOC-3281/#multiple-branches-con
Original Jira: https://issues.jboss.org/browse/BXMSDOC-3281

### CODE REVIEW

File: doc-content/enterprise-only/project-data/create-branches-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[id='create-branches-proc']
+
+= Creating branches
+You can create a new branches in {CENTRAL} and name them whatever you like. Initially, you will only have the default *master* branch. When you create a new branch for a project, you are making a copy of the project and all of its assets. You can then switch to your newly created branch and make changes to the project without impacting the original *master* branch version.
```
Comment (barboras7) [inline doc-content/enterprise-only/project-data/create-branches-proc.adoc]: ... you are making a copy of the project and all of its assets... Although I understand this correctly, this might be a bit confusing for users - because there is an option of "duplicate" for each project, which makes the copy of the project. However, I am not sure how would I explain it better.
 Maybe Paulo could have some suggestions here? :)

Comment (barboras7) [reply doc-content/enterprise-only/project-data/create-branches-proc.adoc]: Also, "you can switch to your newly created branch" - later on, you mention that you are redirected to the newly created branch (which is true). Here, it sounds like you need to switch to it - I would rather say something like "When you are on the newly created branch" or anything like this to avoid the word "switch"

File: doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[id='multiple-branches-con']
+
+= Multiple branches in {CENTRAL}
+
+Multiple branch support in {CENTRAL} provides the ability to copy a project, including all of its assets, to a new branch. All new, imported, and sample projects open in the default *master* branch. You can create as many branches as you need and can work on multiple branches interchangeably without impacting the original project on the *master* branch.
```
Comment (barboras7) [inline doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc]: Here as well (just so that we don't forget about this part) - this also says 'copy a project'

File: doc-content/enterprise-only/project-data/create-branches-proc.adoc (hunk 1 above)
Comment (paulovmr) [reply doc-content/enterprise-only/project-data/create-branches-proc.adoc]: I agree with @barboras7 , I would change "project and all of its assets" to "selected branch and all of its assets" or something like that.

Also there is a typo in the beginning of this line: "You can create a new branches" -> "You can create new branches".

File: doc-content/enterprise-only/project-data/select-branches-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[id='select-branches-proc']
+
+= Selecting branches
+You can switch between branches to review previous project assets and to test new branch projects.
```
Comment (paulovmr) [inline doc-content/enterprise-only/project-data/select-branches-proc.adoc]: I would rephrase this, instead of "to review previous project assets and to test new branch projects", I would use "to test them and review their assets".

Comment (paulovmr) [inline doc-content/enterprise-only/project-data/select-branches-proc.adoc]: I would rephrase this, instead of "to review previous project assets and to test new branch projects", I would use "to test them and review their assets".

File: doc-content/enterprise-only/project-data/create-branches-proc.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/create-branches-proc.adoc]: The previous rewording addresses this IMHO.

File: doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc (hunk 2 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc]: Reworded.

File: doc-content/enterprise-only/project-data/select-branches-proc.adoc (hunk 3 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/select-branches-proc.adoc]: Changed as suggested.

File: doc-content/enterprise-only/project-data/modify-projects-overview-con.adoc [hunk 4]
```diff
@@ -3,4 +3,4 @@
 
 {CENTRAL} includes a number of sample projects that you can use to get familiar with the product and its features. The sample projects have been designed and created to demonstrate a variety of business scenarios. You can modify the sample projects to meet your specific business needs. For example, {PRODUCT} {PRODUCT_VERSION} contains the *Mortgage Process* sample project, which consists of predefined data objects, guided decision tables, guided rules, forms, and a business process. You can edit the sample to refine your business process.
 
-If none of the existing {CENTRAL} project samples align with your requirements, you can import a new project from a git repository. See xref:git-import-project[].
+If none of the existing {CENTRAL} project samples align with your requirements, you can create a new project or import one from a git repository. See xref:git-import-project[].
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/modify-projects-overview-con.adoc]: This phrase seems to mean that importing a project from e git repo is for samples. But importing a project from a git repository is not limited to samples.  A project developed in another BC instance can definitely also be imported, and perhaps also some projects created elsewhere.

File: doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[id='multiple-branches-con']
+
+= Multiple branches in {CENTRAL}
+
+Multiple branch support in {CENTRAL} provides the ability to create a new branch based on an existing one, including all of its assets. All new, imported, and sample projects open in the default *master* branch. You can create as many branches as you need and can work on multiple branches interchangeably without impacting the original project on the *master* branch.
+
+This new functionality adds support for persisting branches, which means that {CENTRAL} remembers the last branch used and will open in that branch when you log back in.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc]: While the description of "new functionality" can be correct for this version, it is a maintenance trap. What if you forget to change this formula and it remains "new" five versions later? I would suggest reformulating to avoid referring to "new functiionality" as this is not the Release Notes. Alternatively mention the version number, hardcoded (this functionality, new in version 7.1...)

File: doc-content/enterprise-only/project-data/new-project-proc.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
+[id='new_project']
+= Creating a project
+
+A project can contain data models, business processes, business rules, and forms that are specific to your project's requirements. When you create a project in {CENTRAL} it is added to the Git repository connected to {CENTRAL}.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/new-project-proc.adoc]: At present, I don't believe you can have a separate "Git repository connected to central" (except by way of git hooks). The project is created in the Git repository built into Central, I think.

File: doc-content/enterprise-only/project-data/new-project-proc.adoc [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+.Procedure
+. Log in to {CENTRAL} and click *Menu* -> *Design* -> *Projects*.
+. Click *Add Project*.
+. Type `anyName` in the *Name* field of the *Add Project* window and click *Add*.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/new-project-proc.adoc]: I think you should make it more clear that it's actually any name not "anyName"

File: doc-content/enterprise-only/project-data/create-branches-proc.adoc [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+. Click *master* -> *Add Branch*.
++
+.Create the new branch menu
+image::getting-started/new-branch.png[Creating the new branch menu]
+
+. Type `testBranch1` in the *Name* field and select *master* in the *From* field of the *Add Branch* window.
+. Click *Add*.
++
+.Add the new branch window
+image::getting-started/test-branch.png[Adding the new branch testBranch1]
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/create-branches-proc.adoc]: I wonder when we do screenshots and when we don't, is there guidance somewhere that we follow?

File: doc-content/enterprise-only/project-data/create-branches-proc.adoc [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+.Procedure
+. Log in to {CENTRAL} and click *Menu* -> *Design* -> *Projects*.
+. Click on the project to create the new branch, for example the *Mortgage Process* sample project.
+. Click *master* -> *Add Branch*.
++
+.Create the new branch menu
+image::getting-started/new-branch.png[Creating the new branch menu]
+
+. Type `testBranch1` in the *Name* field and select *master* in the *From* field of the *Add Branch* window.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/create-branches-proc.adoc]: I think you should make it clearer that "testBranch1" is only an example and you can use any name

File: doc-content/enterprise-only/project-data/select-branches-proc.adoc [hunk 10]
```diff
@@ -0,0 +1,12 @@
+[id='select-branches-proc']
+
+= Selecting branches
+You can switch between branches to test them and review their assets.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/select-branches-proc.adoc]: More importantly you can switch between branches to develop the assets :) And development can go on in each branch separately, or something

File: doc-content/enterprise-only/project-data/delete-branches-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,12 @@
+[id='delete-branches-proc']
+
+= Deleting branches
+You can delete any branch, except for the *master* branch because {CENTRAL} does not allow you to delete the *master* branch to avoid corrupting your environment. You must be in any branch other than *master* for the following procedure to work.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/delete-branches-proc.adoc]: The first sentence here really does not sound right to me, style-wise.

File: doc-content/enterprise-only/project-data/delete-branches-proc.adoc [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
+.Procedure
+. Click image:project-data/dots.png[] in the upper-right corner of the screen and select *Delete Branch*.
++
+.Deleting a branch menu
+image::getting-started/delete-branch-menu.png[Delete a branch menu]
+
+. Input the branch name to be deleted in the *Delete Branch* window and click *Delete Branch*.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/delete-branches-proc.adoc]: Really input and not select?

File: doc-content/enterprise-only/project-data/indexing-branches-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,9 @@
+[id='index-branches-proc']
+
+= Indexing branches
+xxx
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/indexing-branches-proc.adoc]: xxx? but it seems that this file is not currently in the assembly anyway?

File: doc-content/enterprise-only/project-data/select-branches-proc.adoc [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+.Procedure
+. Click on the current branch name menu and select the desired project branch.
++
+.Select a branch menu
+image::getting-started/change-branch.png[Selecting a branch menu]
+
+After selecting the branch, you will be redirected to that branch containing the project and all of the assets that you had defined for that branch.
```
Comment (mramendi) [inline doc-content/enterprise-only/project-data/select-branches-proc.adoc]: I think it's better to do "you are redirected" in this case. Future is appropriate in very limited cases.

File: doc-content/enterprise-only/project-data/create-branches-proc.adoc (hunk 8 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/create-branches-proc.adoc]: I tend to wing it. In using the UI, I go with my own instincts as to some of the less obvious menus etc. Really not sure about the hard and fast rules. I just know that reader prefer them.

File: doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc (hunk 5 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/multiple-branches-project-con.adoc]: Corrected. Thanks!

File: doc-content/enterprise-only/project-data/select-branches-proc.adoc (hunk 10 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/select-branches-proc.adoc]: Modified to clarify.

File: doc-content/enterprise-only/project-data/modify-projects-overview-con.adoc (hunk 4 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/modify-projects-overview-con.adoc]: Corrected.

File: doc-content/enterprise-only/project-data/new-project-proc.adoc (hunk 6 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/new-project-proc.adoc]: Revised to clarify.

File: doc-content/enterprise-only/project-data/new-project-proc.adoc (hunk 7 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/new-project-proc.adoc]: Done.

File: doc-content/enterprise-only/project-data/create-branches-proc.adoc (hunk 9 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/create-branches-proc.adoc]: Done.

File: doc-content/enterprise-only/project-data/delete-branches-proc.adoc (hunk 11 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/delete-branches-proc.adoc]: Yeah, that one sucks. Reworked.

File: doc-content/enterprise-only/project-data/delete-branches-proc.adoc (hunk 12 above)
Comment (michelehaglund) [reply doc-content/enterprise-only/project-data/delete-branches-proc.adoc]: Yup. You actually have to type it in exactly as it is named. No pull-down. I'm suggesting that dev fix this in the UI.
